<a href="https://colab.research.google.com/github/JaimeTorio/UFV_Visualizacion/blob/main/Ejercicios%20de%20clase/%20Clase2b.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Clase 18/10/2024**

In [1]:
import pandas as pd
df = pd.read_csv('datos_ejercicio_ventas.csv')
print(df.head())


         COUNTRY        SUBBRAND  YEAR  MONTH     SCENARIO FORECAST  \
0       Portugal     Lipton (L3)  2023     12  AI_forecast  AI_P02F   
1  Great Britain     Lipton (L3)  2023     12  AI_forecast  AI_P10F   
2          Spain  Pepsi Max (L3)  2023     12  AI_forecast  AI_P09F   
3  Great Britain        7up (L3)  2024     12  AI_forecast  AI_P10F   
4        Hungary     Lipton (L3)  2023      9  AI_forecast  AI_P03F   

   FORECAST_YEAR         AMOUNT  
0         2023.0  754356.237194  
1         2023.0  560030.558029  
2         2023.0   88501.980847  
3         2023.0  363224.511516  
4         2023.0  396176.120491  


In [6]:
# 1. Tamaño del Dataset
dataset_size = df.shape
print(f'Tamaño del dataset: {dataset_size}')

Tamaño del dataset: (18666, 8)


In [7]:
# 2. Número de Actuals y Forecast
actuals_forecast_count = df['SCENARIO'].value_counts()
print(f'Número de Actuals y Forecast: \n{actuals_forecast_count}')

Número de Actuals y Forecast: 
SCENARIO
AI_forecast    17766
actual           900
Name: count, dtype: int64


In [8]:
# 3. Horizontes de predicción (año/mes únicos)
prediction_horizons = df[['YEAR', 'MONTH']].drop_duplicates().shape[0]
print(f'Horizontes de predicción: {prediction_horizons}')

Horizontes de predicción: 29


In [9]:
# 4. Número de países y productos
num_countries = df['COUNTRY'].nunique()
num_products = df['SUBBRAND'].nunique()
print(f'Número de países: {num_countries}')
print(f'Número de productos: {num_products}')

Número de países: 9
Número de productos: 6


In [10]:
# 5. Histórico de datos (Actuals y Forecast)
historical_data_range = df[['YEAR', 'MONTH']].drop_duplicates().sort_values(by=['YEAR', 'MONTH'])
print('Histórico de datos:')
print(historical_data_range.head())

Histórico de datos:
     YEAR  MONTH
34   2023      1
397  2023      2
118  2023      3
9    2023      4
57   2023      5


In [11]:
# 6. Forecasts distintos
distinct_forecasts = df['FORECAST'].nunique()
print(f'Forecasts distintos: {distinct_forecasts}')

Forecasts distintos: 12


1. **Tamaño del Dataset**: 18,666 filas y 8 columnas.
2. **Número de Actuals y Forecast**:
   - AI_forecast: 17,766 registros.
   - Actuals: 900 registros.
3. **Horizontes de predicción**: Hay 29 combinaciones únicas de año y mes (horizontes de predicción).
4. **Número de países**: 9 países.
5. **Número de productos**: 6 productos (submarcas).
6. **Histórico de datos (Actuals y Forecast)**: El rango de datos históricos comienza en enero de 2023.
7. **Forecasts distintos**: Hay 12 tipos de forecast distintos.
